code/game/objects/items/stacks/stack.dm

```diff
@@ -204,7 +204,7 @@
 
 /obj/item/stack/attackby(obj/item/W as obj, mob/user as mob)
 	..()
-	if (istype(W, src.type))
+	if (istype(W, src.type) && src.name==W.name)
```

Comment:
This is a bit of hacky fix.
I'd suggest using if(W.type == src.type) instead, since exact typing would also solve the issue.